c10/core/TensorImpl.h

```diff
@@ -2282,11 +2287,11 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
   }
 
   bool owns_pyobj() {
-    return owns_pyobj_;
+    return static_cast<uintptr_t>(pyobj_) & 1;
   }
 
   void set_owns_pyobj(bool b) {
-    owns_pyobj_ = b;
+    pyobj_ = static_cast<PyObject*>(static_cast<uintptr_t>(_unchecked_untagged_pyobj()) | b);
```

Comment:
nit: AFAICT, there's no guarantee that converting a bool to an int will only produce the value `1` as any non-zero int converts to `true`.

Reply:
" the type bool can be converted to int with the value false becoming ​0​ and true becoming 1. " per https://en.cppreference.com/w/cpp/language/implicit_conversion ; static_cast will hit the implicit conversion per https://en.cppreference.com/w/cpp/language/static_cast item (3) .

Looks like clang makes this work by always ensuring bools have the value 0 or 1 in the first place: https://godbolt.org/z/3W6EqEs4K